### DISCUSSION TITLE
Questions for satellite prometheus fetcher

### DISCUSSION THREAD

**Discussion by dmsolr** [apache/skywalking#7518] [Q&A]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?

I notice that satellite prometheus-fetcher will put some labels(instance, job) to metrics as following. And, satellite will also keep instance label in metadata. 

```
{__name__=apisix_http_status, code=500, instance=localhost:9080, job=prometheus, matched_uri=/2xx, route=t-5xx}
```
In SkyWalking, meter-receiver will also add metadata to labels. Because meter-receiver side builds labels using an immutable hash map, it will throw an exception.

I am confusing whether do we need to put instance and job labels as the metrics labels?
And then do we need to support to rename the label when it is duplication.

**Comment by wu-sheng** [comment]
> I am confusing whether do we need to put instance and job labels as the metrics labels?

I think it makes sense. @gxthrj could confirm. If we don't add it in the fetcher, we will lose this metadata forever. What do you mean meter-receiver could add it?

**Reply by dmsolr** [reply]
Service and ServiceInstance was a part of metadata. `job=prometheus` seems like not a important information. If we need it, I perfer it is tranported by metadata. 
In other word, my question is, could we don't modify metrics' label in fetcher side?
Besides, I'd like to rename the mertics' label in receiver side. Is it acceptable?

**Reply by wu-sheng** [reply]
`Service and ServiceInstance was a part of metadata` there is no guarantee about that. I am not sure what do you mean.

> job=prometheus seems like not a important information

This label is just an example, it is not important. But when you fetch the metrics from 2 kinds of service by using one Satellite instance, such as order and product services, who are going to identify the target?
